src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,244 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INVALID_INDEX_EXCEPTION = "Invalid index: ";
+    private static final String OBJECT_NOT_FOUND_EXCEPTION = "Object not found: ";
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> node = createFirstNode(value);
+            head = node;
+            tail = node;
+            size++;
+        } else {
+            addLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index);
+        if (index == size && index != 0) {
+            addLast(value);
+        } else if (index == 0) {
+            addFirst(value);
```

Comment:
you can improve your condition to simplify it 
```suggestion
        if (index == 0) {
            addFirst(value);
        } else if (index == size) {
            addLast(value);
```